File: back/back/Models/PersonnageMort.cs
using LiteDB;

namespace back.Models;

public sealed class PersonnageMort
{
    [BsonId]
    public int Id { get; set; }

    public string Nom { get; set; } = null!;

    public string DateNaissance { get; set; } = null!;
    public string DateMort { get; set; } = null!;

    public int NbOperation { get; set; }

    public string? ElogeFunebre { get; set; }

    public string? NomGrade { get; set; }

    public string? NomSpecialite { get; set; }
}

File: front/src/models/PersonnageMort.ts
export type PersonnageMortRequete =
{
    id: number,
    nom: string,
    matricule: string,
    dateNaissance: string,
    etatService: string | null,
    groupeSanguin: string,
    estFormateur: boolean,
    estFormateurSpecialite: boolean,
    formationFaite: boolean,
    idSpecialite: number,
    idPlaneteOrigine: number,
    dateMort: string,
    elogeFunebre: string | null
}

export type PersonnageMort2Requete = 
{
    nom: string,
    dateNaissance: string,
    dateMort: string,
    elogeFunebre: string | null,
    nbOperation: number,
    idSpecialite: number,
    idGrade: number
}

export type PersonnageMort =
{
    id: number,
    nom: string,
    dateNaissance: string,
    dateMort: string,
    nbOperation: number,
    nomGrade: string | null,
    nomSpecialite: string | null,
    elogeFunebre: string | null,
}